src/Compilers/CSharp/Portable/Symbols/Synthesized/ReadOnlyListType/SynthesizedReadOnlyListTypeSymbol.cs

```diff
@@ -382,27 +407,18 @@ private SynthesizedReadOnlyListTypeSymbol(SourceModuleSymbol containingModule, s
                     generateNotSupportedException));
             _members = membersBuilder.ToImmutableAndFree();
 
-            // IEnumerable.GetEnumerator()
+            // IEnumerable.GetEnumerator(), IEnumerable<T>.GetEnumerator()
             static BoundStatement generateGetEnumerator(SyntheticBoundNodeFactory f, MethodSymbol method, MethodSymbol interfaceMethod)
             {
                 var containingType = (SynthesizedReadOnlyListTypeSymbol)method.ContainingType;
                 var field = containingType._field;
                 var fieldReference = f.Field(f.This(), field);
-                // return _items.GetEnumerator();
-                return f.Return(
-                    f.Call(
-                        f.Convert(
-                            interfaceMethod.ContainingType,
-                            fieldReference),
-                        interfaceMethod));
-            }
+                if (field.Type.IsTypeParameter())
```

Comment:
>field.Type.IsTypeParameter() [](http://example.com/codeflow?start=20&length=28)

This check is repeated several times. Consider extracting the implementation to a helper, perhaps on `SynthesizedReadOnlyListTypeSymbol`. Perhaps we could store the `SynthesizedReadOnlyListKind` as a field on that type so the check uses that kind.